out of order receiver init

The following program `sample.go` triggers a panic:

```go
package main

func Bar() {
    s := Obj.Foo()
    println(s)
}   

type T struct{}

func (t *T) Foo() bool { return t != nil }

var Obj = NewT()

func NewT() *T { return &T{} }

func main() { Bar() }
```

Expected result:
```console
$ go run ./sample.go
true
```

Got:
```console
$ yaegi ./sample.go
panic: reflect: FieldByName of non-struct type

goroutine 1 [running]:
reflect.(*rtype).FieldByName(0xc0000788a0, 0xc000022766, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/lib/go/src/reflect/type.go:942 +0x135
github.com/containous/yaegi/interp.(*itype).lookupBinField(0xc0001b2dd0, 0xc000022766, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/marc/go/src/github.com/containous/yaegi/interp/type.go:602 +0xa9
github.com/containous/yaegi/interp.(*Interpreter).cfg.func2(0xc0000d0d00)
	/home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1160 +0x39a4
github.com/containous/yaegi/interp.(*node).Walk(0xc0000d0d00, 0xc00019bd68, 0xc00019bda0)
	/home/marc/go/src/github.com/containous/yaegi/interp/interp.go:114 +0xaf
github.com/containous/yaegi/interp.(*node).Walk(0xc0000d0c00, 0xc00019bd68, 0xc00019bda0)
	/home/marc/go/src/github.com/containous/yaegi/interp/interp.go:111 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000d0a00, 0xc00019bd68, 0xc00019bda0)
	/home/marc/go/src/github.com/containous/yaegi/interp/interp.go:111 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000d0900, 0xc00019bd68, 0xc00019bda0)
	/home/marc/go/src/github.com/containous/yaegi/interp/interp.go:111 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000d0400, 0xc00019bd68, 0xc00019bda0)
	/home/marc/go/src/github.com/containous/yaegi/interp/interp.go:111 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc0000d0200, 0xc00019bd68, 0xc00019bda0)
	/home/marc/go/src/github.com/containous/yaegi/interp/interp.go:111 +0x70
github.com/containous/yaegi/interp.(*Interpreter).cfg(0xc0000d6ea0, 0xc0000d0200, 0xc000022748, 0x4, 0x0, 0x0, 0x4)
	/home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:39 +0x1b2
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0000d6ea0, 0xc0001bc000, 0xc0, 0x2c0, 0xc0001bc000, 0xc0, 0x0, 0xc2f220)
	/home/marc/go/src/github.com/containous/yaegi/interp/interp.go:253 +0x135
main.main()
	/home/marc/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:77 +0x2f8
```

This is the exact problem occuring at least in #257

--- a/_test/method21.go
+++ b/_test/method21.go
@@ -13,3 +13,6 @@ func (*Hello) Hi() string {
 func main() {
 	fmt.Println(&Hello{})
 }
+
+// Output:
+// &{}

--- a/_test/method22.go
+++ b/_test/method22.go
@@ -1,0 +1,19 @@
+package main
+
+func Bar() {
+	s := Obj.Foo()
+	println(s)
+}
+
+var Obj = &T{}
+
+type T struct{}
+
+func (t *T) Foo() bool { return t != nil }
+
+func main() {
+	Bar()
+}
+
+// Output:
+// true

--- a/_test/method23.go
+++ b/_test/method23.go
@@ -1,0 +1,21 @@
+package main
+
+func Bar() {
+	s := Obj.Foo()
+	println(s)
+}
+
+var Obj = NewT()
+
+func NewT() *T { return &T{} }
+
+type T struct{}
+
+func (t *T) Foo() bool { return t != nil }
+
+func main() {
+	Bar()
+}
+
+// Output:
+// true

--- a/_test/struct12.go
+++ b/_test/struct12.go
@@ -13,3 +13,6 @@ type S2 struct {
 func main() {
 	fmt.Println(S2{})
 }
+
+// Output:
+// {<nil>}

--- a/_test/struct15.go
+++ b/_test/struct15.go
@@ -21,3 +21,6 @@ func (w GzipResponseWriterWithCloseNotify) CloseNotify() <-chan bool {
 func main() {
 	fmt.Println("hello")
 }
+
+// Output:
+// hello

--- a/_test/struct19.go
+++ b/_test/struct19.go
@@ -21,3 +21,6 @@ func main() {
 	c := CreateConfig()
 	fmt.Println(c)
 }
+
+// Output:
+// &{[]   false }

--- a/_test/type12.go
+++ b/_test/type12.go
@@ -25,3 +25,6 @@ func main() {
 	t := &T1{}
 	println(t.Get())
 }
+
+// Output:
+// no name

--- a/_test/type13.go
+++ b/_test/type13.go
@@ -1,0 +1,12 @@
+package main
+
+var a = &T{}
+
+type T struct{}
+
+func main() {
+	println(a != nil)
+}
+
+// Output:
+// true

--- a/_test/type14.go
+++ b/_test/type14.go
@@ -1,0 +1,14 @@
+package main
+
+import "fmt"
+
+var a = T{}
+
+type T struct{}
+
+func main() {
+	fmt.Println(a)
+}
+
+// Output:
+// {}

--- a/interp/gta.go
+++ b/interp/gta.go
@@ -9,10 +9,11 @@ import (
 // variables and functions symbols at package level, prior to CFG.
 // All function bodies are skipped. GTA is necessary to handle out of
 // order declarations and multiple source files packages.
-func (interp *Interpreter) gta(root *node, rpath string) error {
+func (interp *Interpreter) gta(root *node, rpath string) ([]*node, error) {
 	sc, _ := interp.initScopePkg(root)
 	var err error
 	var iotaValue int
+	var revisit []*node
 
 	root.Walk(func(n *node) bool {
 		if err != nil {
@@ -42,23 +43,24 @@ func (interp *Interpreter) gta(root *node, rpath string) error {
 
 			for i := 0; i < n.nleft; i++ {
 				dest, src := n.child[i], n.child[sbase+i]
-				typ := atyp
 				val := reflect.ValueOf(iotaValue)
+				typ := atyp
 				if typ == nil {
 					if typ, err = nodeType(interp, sc, src); err != nil {
 						return false
 					}
 					val = src.rval
 				}
-				var index int
-				if !typ.incomplete {
-					if typ.cat == nilT {
-						err = n.cfgErrorf("use of untyped nil")
-						return false
-					}
-					index = sc.add(typ)
+				if typ.incomplete {
+					// Come back when type is known
+					revisit = append(revisit, n)
+					return false
 				}
-				sc.sym[dest.ident] = &symbol{kind: varSym, global: true, index: index, typ: typ, rval: val}
+				if typ.cat == nilT {
+					err = n.cfgErrorf("use of untyped nil")
+					return false
+				}
+				sc.sym[dest.ident] = &symbol{kind: varSym, global: true, index: sc.add(typ), typ: typ, rval: val}
 				if n.anc.kind == constDecl {
 					sc.sym[dest.ident].kind = constSym
 					iotaValue++
@@ -171,5 +173,5 @@ func (interp *Interpreter) gta(root *node, rpath string) error {
 	if sc != interp.universe {
 		sc.pop()
 	}
-	return err
+	return revisit, err
 }

--- a/interp/interp.go
+++ b/interp/interp.go
@@ -245,8 +245,14 @@ func (interp *Interpreter) Eval(src string) (reflect.Value, error) {
 	}
 
 	// Global type analysis
-	if err = interp.gta(root, pkgName); err != nil {
+	revisit, err := interp.gta(root, pkgName)
+	if err != nil {
 		return res, err
+	}
+	for _, n := range revisit {
+		if _, err = interp.gta(n, pkgName); err != nil {
+			return res, err
+		}
 	}
 
 	// Annotate AST with CFG infos

--- a/interp/src.go
+++ b/interp/src.go
@@ -33,6 +33,7 @@ func (interp *Interpreter) importSrcFile(rPath, path, alias string) error {
 
 	var initNodes []*node
 	var rootNodes []*node
+	revisit := make(map[string][]*node)
 
 	var root *node
 	var pkgName string
@@ -65,8 +66,20 @@ func (interp *Interpreter) importSrcFile(rPath, path, alias string) error {
 		rootNodes = append(rootNodes, root)
 
 		subRPath := effectivePkg(rPath, path)
-		if err = interp.gta(root, subRPath); err != nil {
+		var list []*node
+		list, err = interp.gta(root, subRPath)
+		if err != nil {
 			return err
+		}
+		revisit[subRPath] = append(revisit[subRPath], list...)
+	}
+
+	// revisit incomplete nodes where GTA could not complete
+	for pkg, nodes := range revisit {
+		for _, n := range nodes {
+			if _, err = interp.gta(n, pkg); err != nil {
+				return err
+			}
 		}
 	}
 

